app/controllers/works_controller.rb

```diff
@@ -0,0 +1,135 @@
+class WorksController < ApplicationController
+
+  before_action :find_work, only: [:destroy, :update, :edit, :show ]
+  before_action :require_login, only: [:new, :create, :update, :edit, :destroy, :upvote ]
+
+  # Helper Methods
+  def not_found_error_notice
+    flash[:notice] = "That media does not exist."
```

Comment:
You should also redirect them here.

```suggestion
    flash[:notice] = "That media does not exist."
    redirect_back(fallback_location: root_path)
```